python/pyproject.toml

```diff
@@ -21,4 +21,9 @@ requires = [
     "scikit-build>=0.13.1",
     "cmake>=3.23.1",
     "ninja",
+    "treelite==3.0.0",
+    "treelite_runtime==3.0.0",
+    "versioneer",
```

Comment:
Versioneer is [vendored](https://github.com/rapidsai/cuml/blob/branch-22.12/python/versioneer.py) in this package. It still explicitly required for the build system? Or should we remove the vendored module?

Reply:
We should remove the vendored module. The vendored approach is no longer needed since versioneer supports the installation method. Additionally, the vendored approach is not compatible with modern Python build backends. I chose not to remove the vendored version because:
1. That also requires modifying the conda recipes and I was trying to limit the scope, and
2. I wanted to make a swathe of PRs that removes it for all RAPIDS packages in 23.02. 

I can remove the vendored module here if you'd prefer, though.

Reply:
> I can remove the vendored module here if you'd prefer, though.

I think totally fine to leave that for a follow-up as long as its tracked somewhere.